### DISCUSSION TITLE
Feature request : Multiple user access management for apps

### DISCUSSION THREAD

**Discussion by gschurck** [caprover/caprover#1315] [Ideas]
It would be very interesting to be able to create multiple users, and each one has only access to some apps, to change its settings and for deployment.

**Comment by githubsaturn** [comment]
This is out of scope due to its subtle complexities. It requires everything on CapRover to be namespaced. Even then, if an application binds to `/var/run/docker.socket` it can control all other apps. 

A more enterprise version of this might be added as a premium feature in the future with auditing and etc. But for CapRover as it stands, it's out of scope. 


As mentioned in the [scope](https://github.com/caprover/caprover/blob/master/CONTRIBUTING.md#important-caprover-goals--scope):
> CapRover is not an enterprise grade application

**Reply by gschurck** [reply]
Ok I see.

**Comment by dkyeremeh** [comment]
@githubsaturn... The multi-user implementation doesn't need to be role based. Everyone can be admin. This is because organisations have users coming in and out. 

If one account is used by everyone, then an ex employee will still have admin access to all the organisation's app. To prevent this, 
 passwords need to change all the time.

@lcx explains it well here: https://github.com/caprover/caprover/issues/492#issuecomment-946644143

**Comment by simonjcarr** [comment]
This request would be useful just for identification. It does not need to provide roles. The main reason for a multiple users feature is so that organizations can track, who, what and when. The fact that all users have admin access is not as issue, and is assumed to be acknowledged when caprover is installed. Once you have users, it's something you can build features on in the future. As mentioned by @dkyeremeh at the moment, every time a user leaves the organisation or changes jobs in the department, the password would need to be changed. Without being disrespectful, for most organisations this limitation makes it a hobby tool, rather than anything that could be incorporated into a commercial production tool chain.

**Comment by sanfx** [comment]
it would have been really useful to have multi user support, even if it is for planned for enterprise version I ready to go for it.
